src/tapedfunction.jl

```diff
@@ -244,12 +234,12 @@ function translate!!(var::IRVar, line::Core.NewvarNode,
                      boxes::Dict{Symbol, Box{<:Any}}, @nospecialize(ir))
     # use a noop to ensure the 1-to-1 mapping from ir.code to instructions
     # on tape. see GotoInstruction.dest.
-    return GotoInstruction(Box{Any}(true), 0)
+    return GotoInstruction(:_true, 0)
 end
 
 function translate!!(var::IRVar, line::GlobalRef,
                      boxes::Dict{Symbol, Box{<:Any}}, @nospecialize(ir))
-    return Instruction(val, (line,), var_boxer(var, boxes))
+    return Instruction(() -> val(line), (), var_boxer(var, boxes))
```

Comment:
Maybe keep the previous implementation since it avoids creating unnecessary anymous function? 
```suggestion
    return Instruction(val, (line,), var_boxer(var, boxes))
```

Reply:
Generally, here `line` is not a Symbol, it may be a `GlobalRef` in most cases, so it can't be put into `Instruction.input` which is a tuple of symbols.